docs/fundamentals/networking/sockets/tcp-classes.md

```diff
@@ -78,10 +78,203 @@ The preceding C# code:
+
+Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient(AddressFamily.InterNetwork);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+```
+
+Other `TcpClient`'s constructor takes an <xref:System.Net.IPEndPoint> class as a parameter. This constructor accepts an `AddressFamily` from <xref:System.Net.IPEndPoint.AddressFamily> property and creates a `Socket`. As part of this, it calls <xref:System.Net.Sockets.Socket.Bind%2A> on the underlying `Socket` member with the given <xref:System.Net.IPEndPoint> argument. Consider the following TCP client code:
+
+```csharp
+// Example IPEndPoint object
+var ep = new IPEndPoint(IPAddress.Parse("127.0.0.1"), 5001);
+var client = new TcpClient(ep);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+// Example IPEndPoint object
+var ep = new IPEndPoint(IPAddress.Parse("127.0.0.1"), 5001);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+socket.Bind(ep);
+```
+
+Another `TcpClient` constructor overload accepts a `hostname` and `port` as parameters, the difference from the default constructor is that this constructor tries to connect to the given `hostname` and `port`. Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient("www.example.com", 80);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+socket.Connect("www.example.com", 80);
+```
+
+### Connect to server
+
+Basically, every `Connect`, `ConnectAsync`, `BeginConnect` and `EndConnect` overload in `TcpClient` are functionally equivalent and their function signatures are same as well. Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient();
+client.Connect("www.example.com", 80);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+socket.Connect("www.example.com", 80);
+```
+
+### Create a server
+
+In `TcpListener` class, there are two constructors. Basically, both of them are initializes underlying socket. One of them is `TcpListener(IPAddress localaddr, int port)`. Consider the following TCP listener code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+```
+
+Other one is `TcpListener(IPEndPoint localEP)`, which does the same thing with the above constructor. Consider the following TCP listener code:
```

Comment:
Demonstrating one constructor should be enough here, they do more or less the same.